### PR TITLE
Updated NCC loss for PyTorch Training

### PR DESCRIPTION
**Pull request by MengjinDong**
Hi voxelmorph researchers,

When I was trying to train voxelmorph for deformable registration with NCC loss with PyTorch (images are affinely registered), I found that the deformation is large and does not converge to a local minimum. Then by comparing the NCC loss of this PyTorch version and the Tensorflow version, I saw there was a slight difference in the way they deal with very small denominators. After updating the PyTorch version to behave the same as the Tensorflow version, now everything works well^.^! 

In this request I modified the NCC loss function in voxelmorph/torch/losses.py and kept everything else the same. I think it would be beneficial to others if this bug is fixed! Thank you!

Best,
-M

### CODE REVIEW

**Comment by kvttt** [comment]
Your edited NCC loss works for me. Thank you!

**Comment by kvttt** [comment]
Hi M,

Thanks for your updated NCC loss. I was just going over your code and I am having some questions about the following line:

`cross = torch.clamp(cross, min = self.eps)`

I think the covariance term, namely cross in the code, can be negative. Therefore, it seems to me that simply clamping the covariance term with a minimum is not reasonable because the purpose of introducing the epsilon is to keep the covariance term away from zero instead of making all the negative terms epsilon. Could you clarify this please? In addition, do you think clamping the covariance term with epsilon is necessary since the covariance term is the nominator? Thank you! A possible fix for this is to clamp the square of cross, i.e. cross * cross with a minimum of epsilon instead of directly clamping cross itself.
